tls/s2n_server_hello.c

```diff
@@ -107,6 +71,53 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
         GUARD(s2n_server_extensions_recv(conn, &extensions));
     }
 
+    if (conn->server_protocol_version >= S2N_TLS13) {
+        /* Check echoed session ID matches */
+        if(session_id_len != conn->session_id_len || !memcmp(session_id, conn->session_id, session_id_len)){
+            GUARD(s2n_queue_reader_illegal_parameter_alert(conn));
```

Comment:
We previously didn't alert for this. Why start?

Reply:
https://tools.ietf.org/html/rfc8446#page-39
RFC specifies it MUST alert with illegal parameter

Reply:
To this point, s2n has skipped sending alerts to keep behavior constant for all different kinds of failures. Historically returning different alert values have been used as oracles in attacks.

In this case, I don't think it's *likely* that returning a specific alert in the case of an invalid session id will be useful to an attacker. But still I think we should stay consistent. I don't think we should start sending alerts here.

Reply:
@raycoll to clarify, are you saying you do not want to check session IDs at all? Or that when we do to use one of the currently used alerts?